### PR TITLE
[Compatibility] Added BLMPOP command

### PR DESCRIPTION
**Pull request by Vijay-Nirmal**
Adding the BLMPOP commands to garnet

- [x] Add BLMPOP commands
- [x] Add Integration Test cases, ACL Test and ~~Slot Verification Test~~ (There is no Blocking command in cluster slot verification test as of now)
- [x] Add documentation

### CODE REVIEW

File: libs/server/Objects/ItemBroker/CollectionItemBroker.cs [hunk 1]
```diff
@@ -412,17 +411,17 @@ private static bool TryGetNextSetObject(SortedSetObject sortedSetObj, RespComman
         /// <param name="command">RESP command</param>
         /// <param name="cmdArgs">Additional command arguments</param>
         /// <param name="currCount">Collection size</param>
-        /// <param name="nextItem">Retrieved item</param>
+        /// <param name="result">Retrieved item</param>
```
Comment (TalZaccai) [inline libs/server/Objects/ItemBroker/CollectionItemBroker.cs:414]: Please update method summary & param comment to reflect the changes

File: libs/server/Objects/ItemBroker/CollectionItemBroker.cs [hunk 2]
```diff
@@ -56,9 +55,9 @@ public class CollectionItemBroker : IDisposable
         /// <param name="timeoutInSeconds">Timeout of operation (in seconds, 0 for waiting indefinitely)</param>
         /// <returns>Result of operation</returns>
         internal async Task<CollectionItemResult> GetCollectionItemAsync(RespCommand command, byte[][] keys,
-            RespServerSession session, double timeoutInSeconds)
+            RespServerSession session, double timeoutInSeconds, ArgSlice[] cmdArgs = null)
```
Comment (TalZaccai) [inline libs/server/Objects/ItemBroker/CollectionItemBroker.cs:58]: Please update XML comment

File: libs/server/Objects/ItemBroker/CollectionItemBroker.cs [hunk 3]
```diff
@@ -501,13 +503,33 @@ private bool TryGetNextItem(byte[] key, StorageSession storageSession, RespComma
...
+                                var _ = ParseUtils.TryReadInt(ref cmdArgs[1], out var popCount); // Int should has be validated in the Network layer (Server Session) itself
+                                popCount = popCount < 1 ? 1 : popCount; // Default when count is not provided
+                                popCount = Math.Min(popCount, listObj.LnkList.Count);
+
+                                var items = new byte[popCount][];
+                                for (var i = 0; i < popCount; i++)
+                                {
+                                    if (!TryGetNextListItem(listObj, popDirection == OperationDirection.Left ? RespCommand.BLPOP : RespCommand.BRPOP, out items[i]))
+                                    {
+                                        return false;
```
[the remark this answers is not in the document]
Comment (Vijay-Nirmal) [reply libs/server/Objects/ItemBroker/CollectionItemBroker.cs:517]: TryGetNextListItem can return false if the list is empty. But not sure if this method can be called when the list is empty. If you are sure then I will do it once you confirm it again.

File: libs/server/Objects/ItemBroker/CollectionItemBroker.cs [hunk 4]
```diff
@@ -501,13 +503,33 @@ private bool TryGetNextItem(byte[] key, StorageSession storageSession, RespComma
                                 }
 
                                 return isSuccessful;
+                            case RespCommand.BLMPOP:
+                                var popDirection = (OperationDirection)cmdArgs[0].ReadOnlySpan[0];
+                                var _ = ParseUtils.TryReadInt(ref cmdArgs[1], out var popCount); // Int should has be validated in the Network layer (Server Session) itself
```
[the remark this answers is not in the document]
Comment (Vijay-Nirmal) [reply libs/server/Objects/ItemBroker/CollectionItemBroker.cs]: I have changed the code like this. I have a question in the below code. Should I pin (fixed) `popCount` variable to take it pointer?

```csharp
// In ListCommands.cs
cmdArgs[1] = new ArgSlice((byte*)&popCount, sizeof(int))

// In CollectionItemBroker.cs
var popCount = *(int*)(cmdArgs[1].ptr);
```

[the remark this answers is not in the document]
Comment (TalZaccai) [reply libs/server/Objects/ItemBroker/CollectionItemBroker.cs]: Indeed, you should pin it, as ArgSlice assumes a pointer to pinned memory.

[the remark this answers is not in the document]
Comment (Vijay-Nirmal) [reply libs/server/Objects/ItemBroker/CollectionItemBroker.cs]: Looks like local variables are already pinned stack so no need to pin it. So we are good with this command
Source: https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0213?f1url=%3FappId%3Droslyn%26k%3Dk(CS0213)

[the remark this answers is not in the document]
Comment (TalZaccai) [reply libs/server/Objects/ItemBroker/CollectionItemBroker.cs]: Ah since popCount is declared locally on that method, yes that makes sense.

**Review by vazois — APPROVED** [review]
Great work!

**Review by vazois — APPROVED** [review]

**Pull request merged**